systems/citybuilding.go

```diff
@@ -5,39 +5,46 @@ import (
 
 	"engo.io/ecs"
 	"engo.io/engo"
+	"engo.io/engo/common"
 )
 
-type MouseTracker struct {
+// ---------------------------------
+// Vars
+
+// CityAssets are the assets for the system
+var CityAssets = []string{"textures/city.png"}
+
+type CityMouseTracker struct {
 	ecs.BasicEntity
-	engo.MouseComponent
+	common.MouseComponent
 }
 
 type City struct {
 	ecs.BasicEntity
-	engo.RenderComponent
-	engo.SpaceComponent
+	common.RenderComponent
+	common.SpaceComponent
 }
 
 type CityBuildingSystem struct {
 	world *ecs.World
 
-	mouseTracker MouseTracker
+	mouseTracker CityMouseTracker
 }
 
-// Remove is called whenever an Entity is removed from the scene, and thus from this system
-func (*CityBuildingSystem) Remove(ecs.BasicEntity) {}
+// ---------------------------------
+// Struct functions
```

Comment:
It has some good qualities to it, especially when used in the `main` file. Nice thinking @Sendoushi :smile: